weblogic-azure-vm/arm-oraclelinux-wls-cluster/arm-oraclelinux-wls-cluster/src/main/arm/nestedtemplates/clusterCustomSSLTemplate.json

```diff
@@ -362,14 +470,40 @@
             "sku": {
                 "name": "[variables('const_storageAccountType')]"
             },
-            "kind": "Storage",
+            "kind": "StorageV2",
             "properties": {
                 "supportsHttpsTrafficOnly": false
             },
             "dependsOn": [
                 "[variables('name_networkSecurityGroup')]"
             ]
         },
+        {
+            "apiVersion": "${azure.apiVersionForPrivateEndpoint}",
+            "name": "[variables('name_privateSaEndpoint')]",
+            "type": "Microsoft.Network/privateEndpoints",
+            "location": "[parameters('location')]",
+            "properties": {
+                  "privateLinkServiceConnections": [
+                     {
+                        "name": "[variables('name_privateSaEndpoint')]",
+                        "properties": {
+                              "privateLinkServiceId": "[resourceId('Microsoft.Storage/storageAccounts/', variables('name_storageAccount'))]",
+                              "groupIds": [
+                                 "file"
+                              ]
+                        }
```

Comment:
@zhengchang907  can you please explain why we need the `privateEndpoints` resource here?

Reply:
@edburns,

I spotted an bug when testing the admin offer in a private VNET. The file share mounting failed because the storage account's endpoint cannot be correctly resolved in this situation. The error msg sample:
```
mount error: could not resolve address for zhengte2227sa.file.core.windows.net: Unknown error
```

After discussed with @galiacheng , the solution to it is creating a private endpoint for the storage account, and use the generated private IP to mount the file share service.

This approach is already applied to admin offer in [this PR](https://github.com/oracle/weblogic-azure/pull/139)